# Limpieza del dataset de bienes raíces

Este es un conjunto de datos (dataset) reales que fue descargado usando técnicas de web scraping. El archivo contiene registros de **Fotocasa**, el cual es uno de los sitios más populares de bienes raíces en España. Contiene miles de datos de casas reales publicadas en la web www.fotocasa.com.

El dataset fue descargado hace algunos años y en ningún caso se obtuvo beneficio económico de ello.

Tu objetivo es extraer tanta información como sea posible con el conocimiento que tienes hasta ahora de ciencia de datos.

Agrega observaciones, conclusiones o notas acerca de los datos obtenidos en cada ejercicio.

In [1]:
import pandas as pd

#### Ejercicio 00

Lee el dataset data/real_estate.csv e intenta visualizar la tabla (★☆☆)

In [2]:
# Este archivo CSV contiene puntos y comas en lugar de comas como separadores
df = pd.read_csv('assets/real_estate.csv', sep=';')
df

Unnamed: 0  id_realEstates  isNew                    realEstate_name  \
0               1       153771986  False            ferrari 57 inmobiliaria   
1               2       153867863  False  tecnocasa fuenlabrada ferrocarril   
2               3       153430440  False                 look find boadilla   
3               4       152776331  False  tecnocasa fuenlabrada ferrocarril   
4               5       153180188  False            ferrari 57 inmobiliaria   
...           ...             ...    ...                                ...   
15330       15331       153901377  False                infocasa consulting   
15331       15332       150394373  False                inmobiliaria pulpon   
15332       15333       153901397  False             tecnocasa torrelodones   
15333       15334       152607440  False                inmobiliaria pulpon   
15334       15335       153901356  False                infocasa consulting   

       phone_realEstate                                       url_inmueble  \
0           912177526.0  https://www.fotocasa.es/es/comprar/vivienda/ma...   
1           916358736.0  https://www.fotocasa.es/es/comprar/vivienda/ma...   
2           916350408.0  https://www.fotocasa.es/es/comprar/vivienda/ma...   
3           916358736.0  https://www.fotocasa.es/es/comprar/vivienda/ma...   
4           912177526.0  https://www.fotocasa.es/es/comprar/vivienda/ma...   
...                 ...                                                ...   
15330       911360461.0  https://www.fotocasa.es/es/comprar/vivienda/ma...   
15331       912788039.0  https://www.fotocasa.es/es/comprar/vivienda/ma...   
15332       912780348.0  https://www.fotocasa.es/es/comprar/vivienda/ma...   
15333       912788039.0  https://www.fotocasa.es/es/comprar/vivienda/ma...   
15334       911360461.0  https://www.fotocasa.es/es/comprar/vivienda/ma...   

       rooms  bathrooms  surface   price  ... level4Id level5Id level6Id  \
0        3.0        2.0    103.0  195000  ...        0        0        0   
1        3.0        1.0      NaN   89000  ...        0        0        0   
2        2.0        2.0     99.0  390000  ...        0        0        0   
3        3.0        1.0     86.0   89000  ...        0        0        0   
4        2.0        2.0    106.0  172000  ...        0        0        0   
...      ...        ...      ...     ...  ...      ...      ...      ...   
15330    2.0        1.0     96.0  259470  ...        0        0        0   
15331    3.0        1.0    150.0  165000  ...        0        0        0   
15332    4.0        2.0    175.0  495000  ...        0        0        0   
15333    3.0        2.0    101.0  195000  ...        0        0        0   
15334    3.0        2.0    152.0  765000  ...        0        0        0   

      level7Id level8Id accuracy          latitude          longitude zipCode  \
0            0        0        0  40,2948276786438  -3,44402412135624     NaN   
1            0        0        1          40,28674           -3,79351     NaN   
2            0        0        0  40,4115646786438  -3,90662252135624     NaN   
3            0        0        0  40,2853785786438  -3,79508142135624     NaN   
4            0        0        0  40,2998774864376  -3,45226301356237     NaN   
...        ...      ...      ...               ...                ...     ...   
15330        0        0        0          40,45416           -3,70286     NaN   
15331        0        0        0          40,36652           -3,48951     NaN   
15332        0        0        0          40,57444           -3,92124     NaN   
15333        0        0        0          40,36967           -3,48105     NaN   
15334        0        0        0          40,45773           -3,69068     NaN   

      customZone  
0            NaN  
1            NaN  
2            NaN  
3            NaN  
4            NaN  
...          ...  
15330        NaN  
15331        NaN  
15332        NaN  
15333        NaN  
15334        NaN  

[15335

## Trabajando con un DataFrame

#### Ejercicio 01

¿Cuál es la casa más cara del dataset? (★☆☆)

Imprime la dirección y el precio de la casa seleccionada. Para visualizar el resultado utiliza un f-string. Por ejemplo:

```py
f'La casa más cara se encuentra en la dirección: {address} y su precio es {price} €'
```

In [9]:
"""casa_cara = df.sort_values(by='price').tail(1)
f"La casa mas cara se encuentra en la direccion: {casa_cara['address'].values[0]} y su precio es de {casa_cara['price'].values[0]} Euros" """

indice_casa_cara = df["price"].idxmax()
casa_cara = df.iloc[df['price'].idxmax()]
f"La casa mas cara se encuentra en la direccion: {casa_cara['address']} y su precio es de {casa_cara['price']} Euros"

'La casa mas cara se encuentra en la direccion: El Escorial y su precio es de 8500000 Euros'

La casa mas cara tiene un valor de 8.5 millones de Euros se encuentra en España precisamente en El Escorial.
Su valor se debe a su gran tamaño y gran cantidad de habitaciones, aunque no se especifican baños ni detalles, en su descrpcion encontramos que es un palacio 

#### Ejercicio 02

¿Cuál es la casa más barata del dataset? (★☆☆)

Imprime la dirección y el precio de la casa seleccionada utilizando f-string

In [ ]:
casa_barata = df.loc[df[df["price"] > 0]["price"].idxmin()]
f"La casa mas barata se encuentra en la direccion: {casa_barata['address']} y su precio es de {casa_barata['price']} Euros"


'La casa mas barata se encuentra en la direccion: Berlin, Coslada y su precio es de 600 Euros'

hemos tenido que filtrar la busqueda de las casas para conseguir la mas barata con un precio, ya que existen casas con su precio a 0, lo cual da a entender que no estan a la venta o no deben tener una estimazion segun su carateisticas. igualmente se podria hacer una investigacion para conseguir su precio basandonos en su zona y calculando el m2, si existen casas en las mismas poblaciones con precio estabelcido.

#### Ejercicio 03

¿Cuál es la casa más grande del dataset? (★☆☆)

Imprime la dirección y el área de las casas seleccionadas utilizando f-string

In [14]:
casa_con_area = df['surface'].notna()
casa_con_area = df[casa_con_area]
casa_grande = df.iloc[df['surface'].idxmax()]
f"La casa mas grande se encuentra en {casa_grande['address']} y su superficie es de {casa_grande['surface']}m2"


'La casa mas grande se encuentra en Sevilla la Nueva y su superficie es de 249000.0m2'

Aca nuevamente se ha tenido que filtrar la busqueda para encontrar casas con valeres de superficie, probablemente se deba a falta de medidas o inclusion de datos.

#### Ejercicio 04

¿Cuál es la casa más pequeña del dataset? (★☆☆)

In [16]:
casa_con_area = df['surface'].notna()
casa_con_area = df[casa_con_area]
casa_pequeña = df.iloc[df['surface'].idxmin()]
f"La casa mas pequeña se encuentra en {casa_pequeña['address']} y su area es de {casa_pequeña['surface']}m2"

'La casa mas pequeña se encuentra en Calle Amparo,  Madrid Capital y su area es de 15.0m2'

#### Ejercicio 05.

¿El dataset contiene valores no admitidos (NAs)? (★☆☆)

- Muestra el nombre de las filas seguidas por un booleano (`True` o `False`) según contengan o no contengan NAs.
- También muestra el nombre de las columnas seguidas por un booleano (`True` o `False`) según contengan o no contengan NAs.

In [23]:
tiene_na = df.isna().any(axis=1)
tiene_na

0        True
1        True
2        True
3        True
4        True
         ... 
15330    True
15331    True
15332    True
15333    True
15334    True
Length: 15335, dtype: bool

In [24]:
tiene_na =  df.isna().any(axis=0)
tiene_na

Unnamed: 0          False
id_realEstates      False
isNew               False
realEstate_name      True
phone_realEstate     True
url_inmueble        False
rooms                True
bathrooms            True
surface              True
price               False
date                False
description          True
address             False
country             False
level1              False
level2              False
level3              False
level4               True
level5              False
level6               True
level7               True
level8               True
upperLevel          False
countryId           False
level1Id            False
level2Id            False
level3Id            False
level4Id            False
level5Id            False
level6Id            False
level7Id            False
level8Id            False
accuracy            False
latitude            False
longitude           False
zipCode              True
customZone           True
dtype: bool

En este ejerciico apreciamos que todas las filas poseen al menos un dato NA a diferencia de las Columnas que es mixto aunque su mayoria no contienen datos NA

#### Ejercicio 06.

Elimina las columnas que contengan NAs del dataset, si aplica (★★☆)

Muestra las dimensiones del DataFrame original y del DataFrame después de las eliminaciones.

In [ ]:
df_sin_todos_na = df.dropna(axis=1, how="all")
f' Columnas totales: {len(df.columns)} columnas totales sin Na {len(df_sin_todos_na.columns)}'

' Columnas totales: 37 columnas totales sin Na 35'

Aqui luego de filtrar las columnas que sean todos sus elementos NA, se eliminan dejando el DF con dos columnas menos, de 37 a 35

#### Ejercicio 07

¿Cuantas poblaciones (columna level5) contiene el dataset? (★☆☆)

- Muestra una lista con los nombres de las poblaciones
- Muestra el total de las mismas

In [28]:
cont_poblaciones = df['level5'].unique()
cont_poblaciones




array(['Arganda del Rey', 'Fuenlabrada', 'Boadilla del Monte',
       'Las Rozas de Madrid', ' Madrid Capital', 'Villaviciosa de Odón',
       'Pinto', 'Valdemoro', 'Navalcarnero', 'Pozuelo de Alarcón',
       'Torrejón de Ardoz', 'Navalagamella', 'San Sebastián de los Reyes',
       'Rivas-vaciamadrid', 'Alpedrete', 'Móstoles',
       'San Fernando de Henares', 'Coslada', 'Becerril de la Sierra',
       'Alcalá de Henares', 'Chinchón', 'Parla', 'Alcorcón',
       'El Escorial', 'Leganés', 'Pedrezuela', 'Majadahonda',
       'Villanueva de la Cañada', 'Villanueva del Pardillo',
       'Torrelodones', 'Moralzarzal', 'Mejorada del Campo', 'Aranjuez',
       'Corpa', 'Getafe', 'Velilla de San Antonio', 'Sevilla la Nueva',
       'San Martín de la Vega', 'Villalbilla', 'Collado Villalba',
       'Alcobendas', 'El Molar (Madrid)', 'Moraleja de Enmedio', 'Algete',
       'Campo Real', 'Torrejón de la Calzada', 'Colmenar Viejo',
       'Valdemorillo', 'Fuente El Saz de Jarama', 'Tres Cantos',

In [29]:
lista_poblaciones = df['level5']
lista_poblaciones.to_list()

['Arganda del Rey',
 'Fuenlabrada',
 'Boadilla del Monte',
 'Fuenlabrada',
 'Arganda del Rey',
 'Las Rozas de Madrid',
 'Las Rozas de Madrid',
 'Las Rozas de Madrid',
 'Las Rozas de Madrid',
 ' Madrid Capital',
 ' Madrid Capital',
 'Villaviciosa de Odón',
 'Pinto',
 'Valdemoro',
 'Navalcarnero',
 ' Madrid Capital',
 ' Madrid Capital',
 ' Madrid Capital',
 'Pozuelo de Alarcón',
 'Torrejón de Ardoz',
 'Navalagamella',
 'San Sebastián de los Reyes',
 ' Madrid Capital',
 'Rivas-vaciamadrid',
 'Alpedrete',
 ' Madrid Capital',
 ' Madrid Capital',
 ' Madrid Capital',
 'Móstoles',
 ' Madrid Capital',
 ' Madrid Capital',
 ' Madrid Capital',
 'San Fernando de Henares',
 'Coslada',
 ' Madrid Capital',
 'Becerril de la Sierra',
 ' Madrid Capital',
 ' Madrid Capital',
 ' Madrid Capital',
 ' Madrid Capital',
 ' Madrid Capital',
 ' Madrid Capital',
 ' Madrid Capital',
 ' Madrid Capital',
 ' Madrid Capital',
 'Alcalá de Henares',
 ' Madrid Capital',
 ' Madrid Capital',
 ' Madrid Capital',
 ' Madrid Ca

In [31]:
f' Son un total de {len(cont_poblaciones)} poblaciones'

' Son un total de 168 poblaciones'

vemos que al realizar la lista de poblaciones nos damos cuenta que hay muchas locaciones repetidas, pero al contarlas solo por valores da un total de 168. lo cual nos simplifica mucho la tarea de buscarlas 

#### Ejercicio 08

¿Cuál es la media de precios en la población (columna level5) de "Arroyomolinos (Madrid)"? (★★☆)

In [33]:
casas_en_arroyomolinos = df[df['level5'] == 'Arroyomolinos (Madrid)']
casas_en_arroyomolinos

Unnamed: 0  id_realEstates  isNew                  realEstate_name  \
453           454       153843083  False                  akris arguelles   
607           608       153620088  False        100 home red inmobiliaria   
613           614       153478908  False               mares inmobiliaria   
707           708       153736961  False                         gm house   
852           853       153274493  False                         gm house   
...           ...             ...    ...                              ...   
15019       15020       152767314  False                 haya real estate   
15089       15090       152767395  False                 haya real estate   
15091       15092       152767377  False                 haya real estate   
15094       15095       152767335  False                 haya real estate   
15192       15193       146831440  False  miranda servicios inmobiliarios   

       phone_realEstate                                       url_inmueble  \
453         911792694.0  https://www.fotocasa.es/es/comprar/vivienda/ma...   
607         912669623.0  https://www.fotocasa.es/es/comprar/vivienda/ma...   
613         911794026.0  https://www.fotocasa.es/es/comprar/vivienda/ma...   
707         916357589.0  https://www.fotocasa.es/es/comprar/vivienda/po...   
852         916357589.0  https://www.fotocasa.es/es/comprar/vivienda/ma...   
...                 ...                                                ...   
15019       911227078.0  https://www.fotocasa.es/vivienda/getafe/calle-...   
15089       911227078.0  https://www.fotocasa.es/es/comprar/vivienda/ar...   
15091       911227078.0  https://www.fotocasa.es/es/comprar/vivienda/ar...   
15094       911227078.0  https://www.fotocasa.es/es/comprar/vivienda/ar...   
15192       912789124.0  https://www.fotocasa.es/es/comprar/vivienda/ma...   

       rooms  bathrooms  surface   price  ... level4Id level5Id level6Id  \
453      2.0        2.0     85.0  165338  ...        0        0        0   
607      4.0        3.0    225.0  251646  ...        0        0        0   
613      4.0        4.0    257.0  396000  ...        0        0        0   
707      3.0        2.0    110.0  225000  ...        0        0        0   
852      4.0        2.0    280.0  373000  ...        0        0        0   
...      ...        ...      ...     ...  ...      ...      ...      ...   
15019    3.0        2.0    143.0  218000  ...        0        0        0   
15089    3.0        2.0    140.0  230000  ...        0        0        0   
15091    3.0        2.0    143.0  217000  ...        0        0        0   
15094    3.0        2.0    142.0  227000  ...        0        0        0   
15192    3.0        2.0    300.0  462000  ...        0        0        0   

      level7Id level8Id accuracy          latitude          longitude zipCode  \
453          0        0        0          40,27657           -3,91864     NaN   
607          0        0        0  40,2568635154027  -3,91214199444133     NaN   
613          0        0        1        40,2839697         -3,9268743     NaN   
707          0        0        0        40,2779047         -3,9136693     NaN   
852          0        0        0        40,2827315         -3,9120264     NaN   
...        ...      ...      ...               ...                ...     ...   
15019        0        0        0       40,27849764        -3,91184112     NaN   
15089        0        0        0       40,27849764        -3,91184112     NaN   
15091        0        0        0       40,27849764        -3,91184112     NaN   
15094        0        0        0       40,27849764        -3,91184112     NaN   
15192        0        0        0         40,270392  -3,90373610000006     NaN   

      customZone  
453          NaN  
607          NaN  
613          NaN  
707          NaN  
852          NaN  
...          ...  
15019        NaN  
15089        NaN  
15091        NaN  
15094        NaN  
15192        NaN  

[137 rows x 37 columns]

In [34]:
f"El promedio de precio de las casas en arroyomolinos es de {int(casas_en_arroyomolinos['price'].mean())}"

'El promedio de precio de las casas en arroyomolinos es de 294541'

Filtramos los datos para enfocarnos solo en una poblacion y asi buscar facilmente su precio promedio. 

#### Ejercicio 09.

¿Los precios promedios de "Valdemorillo" y "Galapagar" son iguales? (★★☆)

- Muestra ambos promedios
- Escribe en una celda markdown una conclusión sobre ellos

In [ ]:
casas_en_valdemorillo = df[df['level5'] == 'Valdemorillo']
casas_en_galapagar = df[df['level5'] == 'Galapagar']
media_valdemorillo = casas_en_valdemorillo['price'].mean()
media_galapagar = casas_en_galapagar['price'].mean()
f' los precios promedio de galapagar son {round(media_galapagar,2)} Euros y de valdemorillo son {round(media_valdemorillo,2)} Euros'

#

' los precios promedio de galapagar son 360063.2 Euros y de valdemorillo son 363860.29 Euros'

Los precios de ambas ubicaciones son diferentes al hacer la comparacion genero un booleano Falso, que implica su desigualdad, ademas a primera vista se aprecia que valdemorillo tiene un valor un poco mas alto 
pero ambas zonas parecen ser similares en costos o cual podria ser similar la inversion en cada una.

#### Ejercicio 10

¿Los promedios de precio por metro cuadrado (precio/m2) de "Valdemorillo" y "Galapagar" son iguales? (★★☆)

> Pista: Crea una nueva columna llamada `pps` (*price per square* o precio por metro cuadrado) y luego analiza los valores.

- Muestra ambos promedios de precio por metro cuadrado
- Escribe en una celda markdown una conclusión sobre ellos

In [38]:
df['pps'] = df['price'] / df['surface']
df

Unnamed: 0  id_realEstates  isNew                    realEstate_name  \
0               1       153771986  False            ferrari 57 inmobiliaria   
1               2       153867863  False  tecnocasa fuenlabrada ferrocarril   
2               3       153430440  False                 look find boadilla   
3               4       152776331  False  tecnocasa fuenlabrada ferrocarril   
4               5       153180188  False            ferrari 57 inmobiliaria   
...           ...             ...    ...                                ...   
15330       15331       153901377  False                infocasa consulting   
15331       15332       150394373  False                inmobiliaria pulpon   
15332       15333       153901397  False             tecnocasa torrelodones   
15333       15334       152607440  False                inmobiliaria pulpon   
15334       15335       153901356  False                infocasa consulting   

       phone_realEstate                                       url_inmueble  \
0           912177526.0  https://www.fotocasa.es/es/comprar/vivienda/ma...   
1           916358736.0  https://www.fotocasa.es/es/comprar/vivienda/ma...   
2           916350408.0  https://www.fotocasa.es/es/comprar/vivienda/ma...   
3           916358736.0  https://www.fotocasa.es/es/comprar/vivienda/ma...   
4           912177526.0  https://www.fotocasa.es/es/comprar/vivienda/ma...   
...                 ...                                                ...   
15330       911360461.0  https://www.fotocasa.es/es/comprar/vivienda/ma...   
15331       912788039.0  https://www.fotocasa.es/es/comprar/vivienda/ma...   
15332       912780348.0  https://www.fotocasa.es/es/comprar/vivienda/ma...   
15333       912788039.0  https://www.fotocasa.es/es/comprar/vivienda/ma...   
15334       911360461.0  https://www.fotocasa.es/es/comprar/vivienda/ma...   

       rooms  bathrooms  surface   price  ... level5Id level6Id level7Id  \
0        3.0        2.0    103.0  195000  ...        0        0        0   
1        3.0        1.0      NaN   89000  ...        0        0        0   
2        2.0        2.0     99.0  390000  ...        0        0        0   
3        3.0        1.0     86.0   89000  ...        0        0        0   
4        2.0        2.0    106.0  172000  ...        0        0        0   
...      ...        ...      ...     ...  ...      ...      ...      ...   
15330    2.0        1.0     96.0  259470  ...        0        0        0   
15331    3.0        1.0    150.0  165000  ...        0        0        0   
15332    4.0        2.0    175.0  495000  ...        0        0        0   
15333    3.0        2.0    101.0  195000  ...        0        0        0   
15334    3.0        2.0    152.0  765000  ...        0        0        0   

      level8Id accuracy          latitude          longitude zipCode  \
0            0        0  40,2948276786438  -3,44402412135624     NaN   
1            0        1          40,28674           -3,79351     NaN   
2            0        0  40,4115646786438  -3,90662252135624     NaN   
3            0        0  40,2853785786438  -3,79508142135624     NaN   
4            0        0  40,2998774864376  -3,45226301356237     NaN   
...        ...      ...               ...                ...     ...   
15330        0        0          40,45416           -3,70286     NaN   
15331        0        0          40,36652           -3,48951     NaN   
15332        0        0          40,57444           -3,92124     NaN   
15333        0        0          40,36967           -3,48105     NaN   
15334        0        0          40,45773           -3,69068     NaN   

      customZone          pps  
0            NaN  1893.203883  
1            NaN          NaN  
2            NaN  3939.393939  
3            NaN  1034.883721  
4            NaN  1622.641509  
...          ...          ...  
15330        NaN  2702.812500  
15331        NaN  1100.000000  
15332        NaN  2828.571429  
15333        NaN  1930.

In [43]:
df_galapagar = df[df['level5'] == 'Galapagar']
df_valdemorillo = df[df['level5'] == 'Valdemorillo']


f' El valor promedio de m2 de galapagar es de: {round(df_galapagar.pps.mean(),2)} y el de valdemorillo es de {round(df_valdemorillo.pps.mean(),2)}'

' El valor promedio de m2 de galapagar es de: 1606.32 y el de valdemorillo es de 1317.95'

Al buscar el precio de metro cuadrado basandonos en surface de cada una de las casas en venta en ambas zonas nos dio como resultado promedios muy diferentes en cada zona siendo el de galapagar el promedio por m2 mas alto.
a la hora de invertir la diferencia por metro puede generar gran diferencia 

#### Ejercicio 11

¿Cuántas agencia de bienes raíces contiene el dataset? (★★☆)

- Muestra el valor obtenido.

In [ ]:
cont_inmobiliarias = df['realEstate_name'].value_counts()
f' hay: {len(cont_inmobiliarias)} inmobiliarias'


' hay: 1821 inmobiliarias'

#### Ejercicio 12

¿Cuál es la población (columna level5) que contiene la mayor cantidad de casas?(★★☆)

- Muestra la población y el número de casas.

In [ ]:
cont_poblaciones = df['level5'].value_counts()
cont_poblaciones.head(1)


level5
Madrid Capital    6643
Name: count, dtype: int64

Madrid capital es la zona con mas casas a la venta, igualmente al ser una zona tan grande lo mejor seria filtrar por niveles mas bajos o localidades mas pequeñas a la hora de querer comprar una casa o comparar precios 

---

## Trabajando con un subconjunto del DataFrame

#### Ejercicio 13

Ahora vamos a trabajar con el "cinturón sur" de Madrid.

Haz un subconjunto del DataFrame original que contenga las siguientes poblaciones (columna level5): "Fuenlabrada", "Leganés", "Getafe", "Alcorcón" (★★☆)

> Pista: Filtra el DataFrame original usando la columna `level5` y la función `isin`.

In [45]:
df_cinturon_sur = df[df['level5'].isin(["Fuenlabrada", "Leganés", "Getafe", "Alcorcón"])]
df_cinturon_sur

Unnamed: 0  id_realEstates  isNew  \
1               2       153867863  False   
3               4       152776331  False   
85             86       153152077  False   
94             95       153995577  False   
109           110       153586414  False   
...           ...             ...    ...   
15275       15276       153903887  False   
15291       15292       151697757  False   
15305       15306       153902389  False   
15322       15323       153871864  False   
15325       15326       153901467  False   

                                 realEstate_name  phone_realEstate  \
1              tecnocasa fuenlabrada ferrocarril       916358736.0   
3              tecnocasa fuenlabrada ferrocarril       916358736.0   
85                       sinergical inmobiliaria               NaN   
94                               viviendas365com       911226014.0   
109              area uno asesores inmobiliarios       912664081.0   
...                                          ...               ...   
15275  aliseda servicios de gestion inmobiliaria       911368198.0   
15291                                    unipiso       912788631.0   
15305                             jadein ferrero       914871639.0   
15322                          gestion comercial       911220662.0   
15325                              montehogar 68       911790675.0   

                                            url_inmueble  rooms  bathrooms  \
1      https://www.fotocasa.es/es/comprar/vivienda/ma...    3.0        1.0   
3      https://www.fotocasa.es/es/comprar/vivienda/ma...    3.0        1.0   
85     https://www.fotocasa.es/es/comprar/vivienda/le...    1.0        1.0   
94     https://www.fotocasa.es/es/comprar/vivienda/le...    3.0        2.0   
109    https://www.fotocasa.es/es/comprar/vivienda/ma...    3.0        3.0   
...                                                  ...    ...        ...   
15275  https://www.fotocasa.es/es/comprar/vivienda/al...    3.0        1.0   
15291  https://www.fotocasa.es/es/comprar/vivienda/al...    3.0        2.0   
15305  https://www.fotocasa.es/es/comprar/vivienda/ma...    3.0        2.0   
15322  https://www.fotocasa.es/es/comprar/vivienda/ma...    3.0        1.0   
15325  https://www.fotocasa.es/es/comprar/vivienda/ma...    2.0        2.0   

       surface   price  ... level5Id level6Id level7Id level8Id accuracy  \
1          NaN   89000  ...        0        0        0        0        1   
3         86.0   89000  ...        0        0        0        0        0   
85        50.0  107000  ...        0        0        0        0        1   
94       120.0  320000  ...        0        0        0        0        0   
109      142.0  425000  ...        0        0        0        0        0   
...        ...     ...  ...      ...      ...      ...      ...      ...   
15275     78.0  138000  ...        0        0        0        0        1   
15291    110.0  279000  ...        0        0        0        0        0   
15305     85.0  170000  ...        0        0        0        0        0   
15322     91.0  112000  ...        0        0        0        0        0   
15325     99.0  215000  ...        0        0        0        0        1   

               latitude          longitude zipCode customZone          pps  
1              40,28674           -3,79351     NaN        NaN          NaN  
3      40,2853785786438  -3,79508142135624     NaN        NaN  1034.883721  
85             40,35059           -3,82693     NaN        NaN  2140.000000  
94             40,31933           -3,77574     NaN        NaN  2666.666667  
109          40,3313411         -3,8313868     NaN        NaN  2992.957746  
...                 ...                ...     ...        ...          ...  
15275          40,31381           -3,83733     NaN        NaN  1769.230769  
15291        40,3259051           -3,76318     NaN        NaN  2536.363636  
15305        40,2882193         -3,8098617     NaN        NaN  2000.000000  
15322          40

#### Ejercicio 14

Calcula la media y la varianza de muestra para las siguientes variables: precio, habitaciones, superficie y baños (★★★)

> Debes usar el subset obtenido en la pregunta 13

- Crea y visualiza un diccionario con todos los valores

In [ ]:
media = df_cinturon_sur[['price','rooms', 'surface','bathrooms']].mean()
media.to_dict()

{'price': 223094.48070562293,
 'rooms': 3.0177383592017737,
 'surface': 111.75222363405337,
 'bathrooms': 1.633221850613155}

In [ ]:
varianza = df_cinturon_sur[['price','rooms', 'surface','bathrooms']].var()
varianza.to_dict()


{'price': 14921367508.04902,
 'rooms': 0.7188858892927542,
 'surface': 4263.051760316337,
 'bathrooms': 0.5717968625577321}

#### Ejercicio 15

¿Cuál es la casa más cara de cada población del cinturón sur de Madríd? (★★☆)

> Debes usar el subset obtenido en la pregunta 13

- Genera un DataFrame con esta información
- Muestra tanto la dirección como el precio de la casa seleccionada de cada población.
- Genera conclusiones en una celda markdown

In [ ]:
casas_caras_pobla_sur = df_cinturon_sur.loc[df_cinturon_sur.groupby('level5')['price'].idxmax()]
casas_caras_pobla_sur[['address','price']]

address    price
5585                                Alcorcón   950000
11561  Calle de Paulo Freire, 5, Fuenlabrada   490000
2881                                  Getafe  1050000
10412           Avenida Reina Sofía, Leganés   650000

Reducimos la lista a nuestro Top por poblacion y vemos que Getafe es la poblacion que tiene la casa mas cara, con precio de 1.05 millones, que aun asi no se acerca a la casa mas cara de toda la muestra que era de 8.5 millones.

#### Ejercicio 16

¿Qué puedes decir acerca del precio por metro cuadrado (precio/m2) entre los municipios de 'Getafe' y 'Alcorcón'? (★★☆)

> Debes usar el subset obtenido en la pregunta 13
>
> Pista: Crea una nueva columna llamada `pps` (price per square en inglés) y luego analiza los valores

In [ ]:
df_getafe = df_cinturon_sur[df_cinturon_sur['level5'] == 'Getafe']
df_alcorcon = df_cinturon_sur[df_cinturon_sur['level5'] == 'Alcorcón']


f' El valor promedio de m2 de Getafe es de: {round(df_getafe.pps.mean(),2)} y el de Alcorcón es de {round(df_alcorcon.pps.mean(),2)}'

' El valor promedio de m2 de getafe es de: 2066.31 y el de Alcorcón es de 2239.3'

Al revisar los datos, en el cinturon sur para las poblaciones especificas de Getafe y Alcorcón, vemos que su precio promedio de m2 no son iguales, 
El precio de Alcorcón esta alrededor de 200 euros por arriba, quizas al encontrar casas con pocos m2 cuadrados no pueda notarse pero cuando sean casas de al menos 100m2 la diferencia 
puede superar los 20 mil euros. 

## Conclusiones

#### Ejercicio 17

El proyecto proporcionó una visión clara de la estructura de precios del mercado inmobiliario, permitiendo identificar las zonas premium y las zonas más asequibles. Las capacidades de Pandas fueron fundamentales para realizar manipulaciones de datos y generar métricas  que ofrecen un panorama completo de las dinámicas de precios y ventas de las propiedades.